compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/PsiUtils.kt

```diff
@@ -111,6 +117,15 @@ public fun KtAnnotated.findAnnotation(fqName: FqName): KtAnnotationEntry? {
     }
   if (hasStarImport) return annotationEntryShort
 
+  // At this point we know that the class is annotated with an annotation that has the same simple
+  // name as fqName. We couldn't find any imports, so the annotation is likely part of the same
+  // package or Kotlin namespace. Leverage our existing utility function and to find the FqName
+  // and then compare the result.
+  val fqNameOfShort = annotationEntryShort.fqNameOrNull(module)
+  if (fqName == fqNameOfShort) {
+    return annotationEntryShort
+  }
```

Comment:
That's the important change. One of my later changes will rely on this. I had to add the module descriptor as argument, what caused the cascading changes.